routes/candidateRoutes.js

```diff
@@ -151,7 +151,7 @@ candidateRouter.put('/update/id=:id', authorizedAdmin, upload.single('file'), as
 candidateRouter.delete('/delete/id=:id', authorizedAdmin, async (req, res) => {
     const candidateId = req.params.id;
 
-    if (!candidateId || candidateId < 0 || Number.isNaN(candidateId)) {
```

Comment:
I don't think you noticed this, but isNaN the global version is different from Number.isNaN, it is not redundant here and you should probably add Number for all other isNaN calls as well.
![image](https://github.com/phamduylong/fp-voter-server/assets/69810988/680b19b4-7490-458f-80d9-35db454a3c2b)

Reply:
Sure. Not sure why strings can be converted to number (I.e. passing the Number.isNaN() method)

Reply:
Where is it converting to number :?

Reply:
Number.isNaN() ?

Reply:
I mean when I am using your code to write the unit tests

Reply:
I can't figure out what you're talking about

Reply:
You used to have this: 
```js
if (!candidateId || candidateId < 0 || Number.isNaN(candidateId))
```
And when I am passing the candidateId as a string (e.g null) it returns 500 instead of 400 (which means it doesn't enter the if route)

Reply:
![image](https://github.com/phamduylong/fp-voter-server/assets/69810988/5de7ee84-a7b1-42e0-ad2e-055e45c33e4e)
It should be
```js
!Number.isNaN(candidateId)
```

Reply:
Oh yeah, somehow I didn't figure it out

Reply:
Right. Looks like Number.isNaN is quite strict and our use cases are unsuitable there. Let's make everything back to global isNaN then. Good catch

Reply:
Sure